sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/properties/AadAuthenticationProperties.java

```diff
@@ -590,27 +592,26 @@ private void validateAuthorizationClientProperties(String registrationId,
                 case WEB_APPLICATION:
                     if (ON_BEHALF_OF.getValue().equals(grantType)) {
                         throw new IllegalStateException("When 'spring.cloud.azure.active-directory.application-type=web_application',"
```

Comment:
private static final String UNMATCHING_OAUTH_GRANT_TYPE_FROMAT = "When 'spring.cloud.azure.active-directory.application-type=%s', 'spring.cloud.azure.active-directory.authorization-clients.%s.authorization-grant-type' can not be '%s'.";